src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -0,0 +1,674 @@
+
+            uint64_t* OrigSrcCid = nullptr;
+            for (uint16_t i = HeaderLength; i < PacketLength; ++i) {
+                if (MagicCid == *(uint64_t*)&Packet[i]) {
+                    OrigSrcCid = (uint64_t*)&Packet[i];
+                }
+            }
+            if (!OrigSrcCid) {
+                printf("Failed to find OrigSrcCid!\n");
+                return;
+            }
+
+            CxPlatRandom(sizeof(uint64_t), DestCid); //fuzz
+            CxPlatRandom(sizeof(uint64_t), SrcCid); //fuzz
+            QUIC_BUFFER* SendBuffer =
+                CxPlatSendDataAllocBuffer(SendData, DatagramLength);
+             if (!SendBuffer) {
+                printf("CxPlatSendDataAllocBuffer failed\n");
+                return;
+                }
+            (*DestCid)++; (*SrcCid)++;
+            *OrigSrcCid = *SrcCid;
+            memcpy(SendBuffer->Buffer, Packet, PacketLength);
+            // printf_buf("cleartext", SendBuffer->Buffer, PacketLength - CXPLAT_ENCRYPTION_OVERHEAD);
+
+            QUIC_PACKET_KEY* WriteKey;
+            
+            if (QUIC_FAILED(
+                QuicPacketKeyCreateInitial(
+                    FALSE,
+                    &HkdfLabels,
+                    InitialSalt.Data,
+                    sizeof(uint64_t),
+                    (uint8_t*)DestCid,
+                 nullptr,
+                    &WriteKey))) {
+                        printf("QuicPacketKeyCreateInitial failed\n");
+                        return;
+                }
+
+            // printf_buf("salt", InitialSalt.Data, InitialSalt.Length);
+            // printf_buf("cid", DestCid, sizeof(uint64_t));
+
+            uint8_t Iv[CXPLAT_IV_LENGTH];
+            QuicCryptoCombineIvAndPacketNumber(
+                WriteKey->Iv, (uint8_t*)&PacketNumber, Iv);
+
+            CxPlatEncrypt(
+                WriteKey->PacketKey,
+                Iv,
+                HeaderLength,
+                SendBuffer->Buffer,
+                PacketLength - HeaderLength,
+                SendBuffer->Buffer + HeaderLength);
+
+            // printf_buf("encrypted", SendBuffer->Buffer, PacketLength);
+
+            uint8_t HpMask[16];
+            CxPlatHpComputeMask(
+                WriteKey->HeaderKey,
+                1,
+                SendBuffer->Buffer + HeaderLength,
+                HpMask);
+
+            // printf_buf("cipher_text", SendBuffer->Buffer + HeaderLength, 16);
+            // printf_buf("hp_mask", HpMask, 16);
+            QuicPacketKeyFree(WriteKey);
+
+            SendBuffer->Buffer[0] ^= HpMask[0] & 0x0F;
+            for (uint8_t i = 0; i < 4; ++i) {
+                SendBuffer->Buffer[PacketNumberOffset + i] ^= HpMask[i + 1];
+            }
+
+            // printf_buf("protected", SendBuffer->Buffer, PacketLength);
+
+        }
+
+        
+        QUIC_SUCCEEDED(
+        CxPlatSocketSend(
+            Binding,
+            &Route,
+            SendData));
+    }
+    printf("All Packets Sent!\n");
```

Comment:
It might be nice to track some stats, such as the number of packets sent, and print those out at the end.